### PR TITLE
style(create-cloudflare): add description support to select prompt

### PR DESCRIPTION
**Pull request by edmundhung**
## What this PR solves / how to test

This PR adds description to each categories and templates with proper text wrapping logic.

Screenshot 2024-08-05 at 12 35 05
Screenshot 2024-08-05 at 12 35 22
Screenshot 2024-08-05 at 12 35 45
Screenshot 2024-08-05 at 12 36 02

## Author has addressed the following

- Tests
  - [x] TODO (before merge)
  - [ ] Included
  - [ ] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required / Maybe required
  - [ ] Not required because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [ ] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Not necessary because: Styles update

### CODE REVIEW

File: packages/create-cloudflare/templates/hello-world-durable-object/c3.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@ export default {
 	configVersion: 1,
 	id: "hello-world-durable-object",
 	displayName: "Hello World Worker Using Durable Objects",
+	description:
```
Comment (jculvey) [inline packages/create-cloudflare/templates/hello-world-durable-object/c3.ts:5]: The template seems a little more bare bones than this to me. Where are the multiplayer application parts in the template?

File: packages/create-cloudflare/src/templates.ts [hunk 2]
```diff
@@ -203,10 +205,27 @@ export const selectTemplate = async (args: Partial<C3Args>) => {
...
 		options: [
-			{ label: "Hello World example", value: "hello-world" },
-			{ label: "Framework Starter", value: "web-framework" },
-			{ label: "Demo application", value: "demo" },
-			{ label: "Template from a Github repo", value: "remote-template" },
+			{
+				label: "Hello World example",
+				value: "hello-world",
+				description:
+					"[Recommended for beginners] Build an empty starter to configure project structure and bare-bones",
```
Comment (jculvey) [inline packages/create-cloudflare/src/templates.ts:212]: I think the copy on these descriptions needs a little more refinement. `Bare-bones starter projects for Workers` or something along those lines might be better here.

File: packages/create-cloudflare/src/templates.ts [hunk 3]
```diff
@@ -203,10 +205,27 @@ export const selectTemplate = async (args: Partial<C3Args>) => {
...
+			},
+			{
+				label: "Framework Starter",
+				value: "web-framework",
+				description: "Create a website with a JavaScript framework",
```
Comment (jculvey) [inline packages/create-cloudflare/src/templates.ts:217]: Perhaps something like `Create an application with a full-stack web framework` might be better

File: packages/create-cloudflare/src/templates.ts [hunk 4]
```diff
@@ -203,10 +205,27 @@ export const selectTemplate = async (args: Partial<C3Args>) => {
...
+			},
+			{
+				label: "Demo application",
+				value: "demo",
+				description: "Build a web application using Cloudflare Workers",
```
Comment (jculvey) [inline packages/create-cloudflare/src/templates.ts:222]: `Build a web application using Cloudflare Workers` -- not all these examples here are web apps, and this doesn't sufficiently distinguish this category from the `hello world` templates. Id probably highlight that the examples here are more full-fledged and showcase advanced usage of the developer products.

**Comment by edmundhung** [comment]
Thanks for the feedback @jculvey! Those descriptions do require some more thoughts. I will share this with Nevi and see how we can make it better.

**Review by jculvey — APPROVED** [review]
Great work!

**Review by RamIdeas — APPROVED** [review]

**Pull request merged**